apps/pwa/src/pages/api/my-events.ts

```diff
@@ -0,0 +1,188 @@
+import type { APIRoute } from "astro";
+import { createClient } from "@supabase/supabase-js";
+import { z } from "zod";
+import { 
+  authHeaderSchema, 
+  createErrorResponse, 
+  userInfoSchema,
+  createSuccessResponse 
+} from "../../lib/schemas";
+
+import { PUBLIC_SUPABASE_URL, PUBLIC_SUPABASE_ANON_KEY } from 'astro:env/client';
+
+export const GET: APIRoute = async ({ request }) => {
+  try {
+    // Validate auth header
+    const authHeader = request.headers.get("Authorization");
+    if (!authHeader) {
+      return createErrorResponse("Authentication required", 401);
+    }
+    
+    authHeaderSchema.parse(authHeader);
+    const token = authHeader.substring(7);
+
+    // Get user info from token
+    let userId: string | undefined;
+    const tokenParts = token.split(".");
+
+    if (tokenParts.length === 3) {
+      // JWT token - decode it
+      try {
+        const payload = tokenParts[1];
+        const base64 = payload.replace(/-/g, "+").replace(/_/g, "/");
+        const padded = base64 + "==".substring(0, (4 - base64.length % 4) % 4);
+        const decoded = atob(padded);
+        const tokenPayload = JSON.parse(decoded);
+        userId = tokenPayload.sub;
+      } catch (error) {
+        console.error("Error decoding JWT token:", error);
+      }
+    } else {
+      // Opaque token - fetch user info from Logto
+      try {
+        const userinfoResponse = await fetch(
+          "https://auth.gocebby.com/oidc/me",
+          {
+            headers: {
+              "Authorization": `Bearer ${token}`,
+            },
+          },
+        );
+
+        if (userinfoResponse.ok) {
+          const rawUserInfo = await userinfoResponse.json();
+          const userInfo = userInfoSchema.parse(rawUserInfo);
+          userId = userInfo.sub;
+        }
+      } catch (error) {
+        console.error("Error fetching user info:", error);
+      }
+    }
+
+    if (!userId) {
+      return createErrorResponse("Could not retrieve user information", 400);
+    }
```

Comment:
This is repeated across 3 files. Can we refactor this to its own file and re-use?